File: app/src/main/java/com/simcom/ecashier/ui/history/CollectionsRecyclerViewAdapter.java
package com.simcom.ecashier.ui.history;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.simcom.ecashier.R;
import com.simcom.ecashier.model.room.Collection;
import com.simcom.ecashier.ui.currentCollection.CollectionRecyclerViewAdapter;

import java.util.ArrayList;
import java.util.List;

public class CollectionsRecyclerViewAdapter extends RecyclerView.Adapter<CollectionsRecyclerViewAdapter.ViewHolder> {

    private List<Collection> collections = new ArrayList<>();

    public void setCollections(List<Collection> collections) {
        this.collections = collections;
        notifyDataSetChanged();
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_collection,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return collections.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder{


        public ViewHolder(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/simcom/ecashier/ui/history/CollectionsRecyclerViewAdapter.kt
package com.simcom.ecashier.ui.history

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.simcom.ecashier.R
import com.simcom.ecashier.model.room.Collection

class CollectionsRecyclerViewAdapter(private var collections: List<Collection>) :
    RecyclerView.Adapter<CollectionsRecyclerViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_collection, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.itemView.findViewById<TextView>(R.id.name).text = collections[position].name
    }
    override fun getItemCount(): Int {
        return collections.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}